PhysicsTools/NanoAOD/python/nano_cff.py

```diff
@@ -218,7 +218,7 @@ def nanoAOD_recalibrateMETs(process,isData):
         from RecoJets.JetProducers.ak4PFJets_cfi import ak4PFJets
         from PhysicsTools.PatAlgos.tools.helpers import getPatAlgosToolsTask, addToProcessAndTask
         task = getPatAlgosToolsTask(process)
-        addToProcessAndTask('ak4PuppiJets', ak4PFJets.clone (src = 'puppi', doAreaFastjet = True, jetPtMin = 2.), process, task)
+        addToProcessAndTask('ak4PuppiJets', ak4PFJets.clone (src = 'puppi', doAreaFastjet = True, jetPtMin = 15.), process, task)
         from PhysicsTools.PatAlgos.tools.jetTools import addJetCollection
```

Comment:
this change goes in the right direction and is compatible on what is done with the TYPE1 in general
https://github.com/cms-sw/cmssw/blob/60e3a2a7a953d787bff64db76132db2e623359a8/JetMETCorrections/Type1MET/python/correctionTermsPfMetType1Type2_cff.py#L40-L45

besides this puppiV15 need of reclustering, how the EM and Muon like objects are handled in nano ?

Reply:
Hello, the EM and muon cleaning are performed inside the runMetCorAndUncFromMiniAOD call which takes as input the uncleaned jet collection, so things are as expected.